package.json

```diff
@@ -118,6 +118,10 @@
   "workspaces": [
     "integration-tests"
   ],
+  "jest": {
+    "clearMocks": true,
+    "resetMocks": false
```

Comment:
If we do this, I will need to look at existing tests and make sure they reset mocks to avoid unintentional impact to other tests (`afterAll`?).

I think I'd rather tests proactively indicate which mocks they depend on. I don't think it's a big burden to have `beforeEach`.

Reply:
+1

Reply:
I generally agree with the beforeEach argument.  The tricky part is when we start using more shared mocks... the pattern I plan to socialize for shared mocks will have similar discoverability and explicit usage as the "beforeEach" pattern,  but align better with the top-of-the-test-file module mocking mechanic... and these new settings (specifically resetMocks: false) are required for that flow to work as expected.

Reply:
@msilva-broad why can't the shared mocks just be instantiated in `beforeEach`?

Reply:
at a minimum, the module being mocked must be called with jest.mock('module-path]) to hoist the fact that it's mocked/mockable to the jest test harness.  And that must be done near the top of the file.  At that point, yes, you can remock it in beforeEach.  And maybe that's fine.  

The pattern I implemented and will be proposing achieves type-safety and discoverability (goto-definition, find-all-references) and tries to do it in one step with something like jest.mock('module-path', () => { const sharedMock = requireActual('shared-mock-path)  return sharedMock.mock(); }.  And that one code snippet is easily cut-and-paste-able to the various unit testing files that want to use it.

Reply:
Could the hoisting be done in a helper file that the test then imports?

Reply:
unfortunately I am pretty certain the answer is no... the hoisting is a very specific jest thing with how it pre-processes test files... it must happen at the top of the file containing the unit tests.

Reply:
discussed this with @nawatts , and I am recommending we delay the resetMocks change until after the work on shared jest mocks is underway and we have a more concrete sense of recommended patterns, etc.